src/database/database.ts

```diff
@@ -0,0 +1,6 @@
+export interface Database {
```

Comment:
we don't need an interface

Reply:
Any reason not to though? I feel like having an interface would be good for future-proofing in case we ever want to swap out the database. Plus if we ever want to write tests that use a mock database, we can program up a mock database that implements this db.

"Program to an interface, not an implementation"

Reply:
the interface is dependent upon the way firebase is implemented and wont translate well to any other db implementation. We don't code to the interface now and there really isnt a need either. I see your point though.

Reply:
I would disagree about the database interface being coupled with Firebase. It takes in collection names, document IDs, and objects to be stored/updated, and they return Observables, which is coupled to our web app, not to the database. At the least, this would make it easy to swap Firebase with an API (since connecting anything else to a front-end would be ludicrous).

I get how it doesn't provide any additional benefit at the moment, but later on when we make calls to the DB from an epic, the database object we use is just going to need CRUD methods, which this interface contains. I personally think it's better OOP practice to have the interface here, and later on, when we write tests for the epics that load and store data, we can just implement the Database interface to make a mock database that has the CRUD methods. I get that we can mock out the database without the interface too, but idk, I guess I like leveraging the OOP-style of Typescript.

But anyhow, I don't see any significant detriment to having the interface since it doesn't create any additional testable code.

@joeySchentrup , your thoughts?

Reply:
I see your point. I argue that since we don't use the interface anywhere or test against it, it should remain out of the code base until we are able to use it. By principle of minimality, we should keep the codebase only full of code we use. I am ok with keeping this file only if we actually use it. Saying we ~might~ use it later isn't a good enough reason.

Also considering my point on it being coupled to firebase: You are right that in the OOP perspective it isn't tied to it; however, what I'm saying is that the very way we wrote the api itself expects firebase as a backend. If we intend to use a future proof soltion, then we would use something that isn't tied like this.